packages/esm-generic-patient-widgets-app/src/resources/useObs.ts

```diff
@@ -18,11 +18,41 @@ export function useObs(patientUuid: string): UseObsResult {
   `,
     openmrsFetch,
   );
+
+  //Buscar conceito para a resposta
+
   const observations =
-    result?.data?.entry?.map((entry) => ({
-      ...entry.resource,
-      conceptUuid: entry.resource.code.coding.filter((c) => isUuid(c.code))[0]?.code,
-    })) ?? [];
+    result?.data?.entry?.map((entry) => {
+      let dataEntry: any = {};
+
+      switch (true) {
```

Comment:
This would be better written as a series of `if... else` statements, I think. also, there's a fair bit of shared implementation, e.g., this could be done as:

```ts
result?.data?.entry?.map((entry) => {
			const observation: ObsResult = {
				...entry.resource,
	      conceptUuid: entry.resource.code.coding.filter((c) => isUuid(c.code))[0]?.code
			};
      
			if (entry.resource.hasOwnProperty("valueString")) {
        observation.dataType = "Text"
      }
```

Or something similar.